lib/pbench/server/api/resources/auth.py

```diff
@@ -0,0 +1,94 @@
+import jwt
+import os
+import datetime
+from flask import session, request, abort
+from flask_httpauth import HTTPTokenAuth
+from pbench.server.api.resources.models import UserModel
+
+
+auth = HTTPTokenAuth("Bearer")
+
+
+def encode_auth_token(token_expire_duration, user_id, logger):
+    """
+    Generates the Auth Token
+    :return: string
+    """
+    payload = {
+        "iat": datetime.datetime.utcnow(),
+        "exp": datetime.datetime.utcnow()
+        + datetime.timedelta(minutes=int(token_expire_duration)),
+        "sub": user_id,
+    }
+    try:
+        # Get jwt key
+        jwt_key = get_secret_key()
+        return jwt.encode(payload, jwt_key, algorithm="HS256")
+    except Exception as e:
+        logger.info(
+            f"Exception occurred while encoding the auth token for user_id {user_id}, exception: {e}"
+        )
+        return None
+
+
+@auth.verify_token
+def verify_token(auth_token):
+    """
+    Validates the auth token
+    :param auth_token:
+    :param app:
+    :return: integer|string
+    """
+    try:
+        payload = jwt.decode(auth_token, get_secret_key(), algorithms="HS256")
+        user_id = payload["sub"]
+        if user_id not in session:
+            return False
+        if auth_token not in session[user_id]:
+            return False
+        user = UserModel.query.filter_by(id=user_id).first()
+        return user
+    except jwt.ExpiredSignatureError:
+        print("Signature expired")
+        return False
```

Comment:
@dbutenhof yes but while trying to implement that I realized that to remove the session we need a `user_id` (since its a session key) and this exception will happen when try to decode the expired session. And if we cant decode the expired session we can't get the user_id since its encoded inside payload and we cant get the payload since we cant decode it. So we cant remove the expired session token here, we would need other way to do that